components/script/textinput.rs

```diff
@@ -442,12 +442,24 @@ impl<T: ClipboardProvider> TextInput<T> {
         let target_line: isize = self.edit_point.line as isize + adjust;
 
         if target_line < 0 {
-            self.edit_point.index = 0;
-            self.edit_point.line = 0;
+            let zero_point = TextPoint { index: 0, line: 0 };
+            self.edit_point = zero_point.clone();
```

Comment:
Can we keep the original field assignments instead of cloning the zero point? That seems to be more efficient to me.

Reply:
Ok